Add pyrefly basic support

stroxler opened the pull request:
I'm just following the example from red-knot.

It seems to work, I developed using
```
mypy_primer -k janus --type-checker pyrefly --debug
```
and once all the crashes were fixed I got a handful of errors as expected.

I also ran
```
mypy_primer --type-checker pyrefly
```
which goes a long while and dumps quite a lot of errors since Pyrefly has a long way to go, but it seemed to run fine through to the end.

hauntsaninja commented on pyproject.toml:
```diff
@@ -13,6 +13,10 @@ classifiers = [
     "Topic :: Software Development",
 ]
+dependencies = [
+    "ipython>=8.35.0",
```
stray?

hauntsaninja commented on mypy_primer/model.py:
```diff
@@ -344,6 +347,63 @@ async def run_knot(
+        if prepend_path is not None:
+            env["MYPY_PRIMER_PREPEND_PATH"] = str(prepend_path)
+
+        env["CLICOLOR_FORCE"] = "1"
```
can remove if pyrefly doesn't use this env var

hauntsaninja reviewed (approved):
Looks good!

stroxler commented:
Addressed code review points.

I also realized the `--typeshed` flag was copypasta, Pyrefly currently cannot pick up a typeshed at runtime because the standard library stubs get injected directly into the binary

It should be possible to run mypy_primer on typeshed changes, but we have to inject the typeshedby overwriting the vendored typeshed at build time which seems like it should wait for a separate PR
